plugins/languages/rust/rustaceanvim/default.nix

```diff
@@ -78,18 +78,9 @@ helpers.neovim-plugin.mkNeovimPlugin config {
       # If nvim-lspconfig is enabled:
       (mkIf config.plugins.lsp.enable {
         # Use the same `on_attach` callback as for the other LSP servers
-        plugins.rustaceanvim.settings.server.on_attach = mkDefault "__lspOnAttach";
-
-        # Ensure the on_attach config is placed **after** the rest of the LSP configuration
-        # (and thus after the declaration of `__lspOnAttach`)
-        plugins.lsp.postConfig = ''
-          do
-            local rustaceanvim = vim.g.rustaceanvim
-            if rustaceanvim[server] == nil then
-              rustaceanvim.server = {}
-            end
-            rustaceanvim.server.on_attach = ${helpers.toLuaObject cfg.settings.server.on_attach}
-            vim.g.rustaceanvim = rustaceanvim
+        plugins.rustaceanvim.settings.server.on_attach = mkDefault ''
+          function()
+            return __nixvim.lspOnAttach
           end
```

Comment:
I think we'd need to invoke the function here instead of returning it

```lua
function(client, bufnr)
  __nixvim.lspOnAttach(client, bufnr)
end
```

Depending on how lazy lua is, I guess we _might_ be able to do that without the new `__nixvim` table? I.e. if lua doesn't evaluate the function body until it is called... Not sure.